src/toil/jobStores/aws/jobStore.py

```diff
@@ -1096,14 +1154,21 @@ def save(self):
                 self._previousVersion = self._version
                 if numNewContentChunks < self._numContentChunks:
                     residualChunks = range(numNewContentChunks, self._numContentChunks)
-                    attributes = [self._chunkName(i) for i in residualChunks]
+                    residual_chunk_names = [self._chunkName(i) for i in residualChunks]
+                    # boto3 requires providing the value as well as the name in the attribute, and we don't store it locally
```

Comment:
I don't think this is true on the backend. The PHP bindings were changed to not need it: https://github.com/aws/aws-sdk-php/issues/185

Is this a boto3 bug? What happens if you send a value different than what is stored, such as an empty string? Or if you don't actually send the value? Maybe the docs are wrong that value is required?

Reply:
I think the docs are wrong in this case. I was using the stubs/boto3 docs when originally looking at this; Pycharm thinks the code is wrong as omitting the "Value" member of the dictionary means the argument type is no longer `List[AttributeTypeDef]`

Reply:
It might just be a quirk of boto3 as the `AttributeNames` argument must match a specific typed dict.
```
botocore.exceptions.ParamValidationError: Parameter validation failed:
Missing required parameter in Attributes[0]: "Value"
```

Reply:
Does it matter what we pass there? Does it still work if we don't bother to get the real current values and always pass an empty string? Or does it get treated like an expect and not do the delete if the value isn't the one that's there?

Reply:
If we pass the empty string, I think it tries to delete an attribute with an empty string value.
```python
>>> response = sdb.put_attributes(DomainName=domain_name, ItemName="test-item", Attributes=[{"Name": "test-name", "Value": "test-value", "Replace": True}])
>>> response = sdb.get_attributes(DomainName=domain_name, ItemName="test-item", AttributeNames=["test-name"])
>>> response
{'Attributes': [{'Name': 'test-name', 'Value': 'test-value'}], 'ResponseMetadata': {'RequestId': '1a3a63b6-fc69-43a1-a25e-e00310ef7f39', 'BoxUsage': '0.0000093222', 'HTTPStatusCode': 200, 'HTTPHeaders': {'date': 'Fri, 26 Apr 2024 18:39:23 GMT', 'content-type': 'text/xml', 'transfer-encoding': 'chunked', 'connection': 'keep-alive', 'vary': 'Accept-Encoding', 'server': 'Amazon SimpleDB'}, 'RetryAttempts': 0}}
>>> response = sdb.delete_attributes(DomainName=domain_name, ItemName="test-item", Attributes=[{"Name": "test-name", "Value": ""}])
>>> response
{'ResponseMetadata': {'RequestId': 'fe5d7a31-c52f-497a-df0a-4176aaf13d0b', 'BoxUsage': '0.0000219909', 'HTTPStatusCode': 200, 'HTTPHeaders': {'date': 'Fri, 26 Apr 2024 18:40:40 GMT', 'content-type': 'text/xml', 'transfer-encoding': 'chunked', 'connection': 'keep-alive', 'vary': 'Accept-Encoding', 'server': 'Amazon SimpleDB'}, 'RetryAttempts': 0}}
>>> response = sdb.get_attributes(DomainName=domain_name, ItemName="test-item", AttributeNames=["test-name"])
>>> response
{'Attributes': [{'Name': 'test-name', 'Value': 'test-value'}], 'ResponseMetadata': {'RequestId': '13fa4c91-c855-3d88-c1d0-bf576bb568c0', 'BoxUsage': '0.0000093222', 'HTTPStatusCode': 200, 'HTTPHeaders': {'date': 'Fri, 26 Apr 2024 18:40:51 GMT', 'content-type': 'text/xml', 'transfer-encoding': 'chunked', 'connection': 'keep-alive', 'vary': 'Accept-Encoding', 'server': 'Amazon SimpleDB'}, 'RetryAttempts': 0}}
>>> response = sdb.delete_attributes(DomainName=domain_name, ItemName="test-item", Attributes=[{"Name": "test-name", "Value": "test-value"}])
>>> response = sdb.get_attributes(DomainName=domain_name, ItemName="test-item", AttributeNames=["test-name"])
>>> response
{'ResponseMetadata': {'RequestId': 'cf383b93-220f-7a7a-b6de-561792180d1e', 'BoxUsage': '0.0000093202', 'HTTPStatusCode': 200, 'HTTPHeaders': {'date': 'Fri, 26 Apr 2024 18:41:04 GMT', 'content-type': 'text/xml', 'transfer-encoding': 'chunked', 'connection': 'keep-alive', 'vary': 'Accept-Encoding', 'server': 'Amazon SimpleDB'}, 'RetryAttempts': 0}}
```

Reply:
OK, then I guess we do need the extra round trip.